fe/fe-core/src/main/java/org/apache/doris/catalog/ScalarType.java

```diff
@@ -291,7 +291,8 @@ public static ScalarType createVarcharType(String lenStr) {
     public static ScalarType createStringType() {
         // length checked in analysis
         ScalarType type = new ScalarType(PrimitiveType.STRING);
-        type.len = -1;
+        // 2GB - 4
+        type.len = 2147483643;
```

Comment:
Define this magic number some where. Maybe in Type.java?
And explain why minus 4 in comment.